File: examples/Python/calculator.py
import unittest
from selenium import webdriver
#pip install eyes-selenium==3.5
from applitools.eyes import Eyes 

class SimpleCalculatorTests(unittest.TestCase):

	def setUp(self):
		desired_caps = {}
		desired_caps["app"] = "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App"
		self.driver = webdriver.Remote(command_executor='http://127.0.0.1:4723', desired_capabilities= desired_caps)
		self.eyes = Eyes()
		self.eyes.api_key = "your_applitools_key"
		self.driver.set_window_size(700, 700) #set size for responsive design testing...

	def tearDown(self):
		self.driver.quit()
		self.eyes.abort_if_not_closed()

	def test_initialize(self):
		self.eyes.open(driver = self.driver, app_name = 'Windows Calculator', test_name = 'Python Windows Calculator Test')
		self.driver.find_element_by_name("Clear").click()
		self.driver.find_element_by_name("Seven").click()
		self.eyes.check_window("Displays 7")
		self.eyes.close()

	def test_addition(self):
		self.eyes.open(driver = self.driver, app_name = 'Windows Calculator', test_name = 'Python Windows Calculator Test')
		self.driver.find_element_by_name("Clear").click()
		self.driver.find_element_by_name("One").click()
		self.driver.find_element_by_name("Plus").click()
		self.driver.find_element_by_name("Seven").click()
		self.driver.find_element_by_name("Equals").click()
		self.eyes.check_window("Displays 8")
		self.eyes.close()

if __name__ == '__main__':
    suite = unittest.TestLoader().loadTestsFromTestCase(SimpleCalculatorTests)
    unittest.TextTestRunner(verbosity=2).run(suite)

File: examples/Ruby/calculator.rb
#https://www.maketecheasier.com/access-windowsapps-folder-windows-10/
#https://github.com/Microsoft/WinAppDriver

gem 'eyes_selenium', '=3.10.1'
require 'eyes_selenium'
require 'rspec'

describe 'Calculator App' do

  before :all do
    caps = { app: "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App" }
    browser = Selenium::WebDriver.for(:remote, :url => "http://127.0.0.1:4723/", :desired_capabilities => caps )
    @eyes = Applitools::Selenium::Eyes.new
    @eyes.api_key = "your_applitools_key"
    @driver = @eyes.open(driver: browser, app_name: "Calculator App", test_name: "Ruby Calculator Test")
    @driver.manage.window.resize_to(700, 700) #For responsive design testing...
  end
  
  after :each do
    @eyes.abort_if_not_closed
    @driver.quit
  end
  
  it 'Addition' do
    @driver.find_element(:name, "Clear").click
    @driver.find_element(:name, "One").click
    @driver.find_element(:name, "Plus").click
    @driver.find_element(:name, "Seven").click
    @driver.find_element(:name, "Equals").click
    @eyes.check_window "equals 8"
     
    results = @eyes.close(false)
    expect(results.passed?).to eq true
  end
end